global: backports Flask-IIIF extension from 2.1 to 2.0

blixhavn opened the pull request.

blixhavn commented:
Added `Amended-by` as I didn't feel any of the other fits the purpose. Should I change it?

tiborsimko commented:
This PR backports a new feature into an already-released maintenance branch.  Two general comments: (1) it would be preferred to fix any bugs you spotted in v2.1 so that you could use the latest version instead; (2) if not possible, and you'd need to backport a feature, then it should be optional and not switched on by default, so that other installations aren't affected...

blixhavn commented:
1) Unfortunately, there are too many large and critical bugs in 2.1 to make it usable in production. (e.g. search with special characters not working [https://github.com/inveniosoftware/invenio/issues/3296], facets not working [https://github.com/inveniosoftware/invenio/issues/3333], the collection admin interface is lacking basic functionality)

2) As per our talk, I have removed the addition of `flask_iiif` to the extensions list to avoid adding features to existing 2.0 installations. In addition, I have now removed Flask-IIIF from pip requirements.

jirikuncar commented:
> I have now removed Flask-IIIF from pip requirements.

You can add it as optional extra (e.g. `iiif`).

jirikuncar commented:
> 15: M102 unrecognized bullet/signature

Please use one of our agreed signatures. We should also remove the bullets with label `NEW`.

blixhavn commented:
There, `iiif` added as optional extra, and commit message fixed.

jirikuncar commented:
@blixhavn can you mention sha1 of the "Original commit".

WDYT about following commit message:
```
* NOTE Backports original commit <sha1>. The extension is not enabled by default in order to avoid ...
```

blixhavn commented:
Looks good. Amended.

jirikuncar commented on setup.py:
```diff
@@ -199,6 +199,9 @@ def run(self):
     "github": [
         "github3.py>=0.9"
     ],
+    "iiif": [
```
Can you please put it before `img`?  We try to keep it sorted. Thanks

jirikuncar commented:
After https://github.com/inveniosoftware/invenio/pull/3338#discussion_r34462813 and rebase we can :shipit:

jirikuncar commented:
@blixhavn please finish sentence with a dot. 

```
* NOTE Backports original commit
  <sha1>. The extension is not ...
```

jirikuncar commented:
@blixhavn `s/orig\./original/` + rebase

blixhavn commented:
`original` was shortened to allow for the full sha1 within 72 characters. Should write `original` and add a new line before the sha1?

jirikuncar commented:
@blixhavn see https://github.com/inveniosoftware/invenio/pull/3338#issuecomment-120932651 ;)

blixhavn commented:
Which sentence is not finished with dot now?

jirikuncar commented:
```
* NOTE Backports original commit
  213b6f1144734c9ecf425a1bc7b78e56ee5e4e3e.  The extension is not 
  enabled by default in order to avoid feature addition to existing
  minor release.
```

jirikuncar commented:
```
* NOTE Backports Flask-IIIF extension from original commit ...
```
